### PR TITLE
Add sample for baremetal deployment

### PR DESCRIPTION
**Pull request by rabi**
Add dataplane sample for deployment with baremetal provisioning. We will kustomize and use this in the jobs.

### CODE REVIEW

**Comment by rabi** [comment]
recheck

**Review by fao89 — APPROVED** [review]

**Comment by fao89** [comment]
recheck

File: config/samples/dataplane_v1beta1_openstackdataplane_baremetal.yaml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      node:
+        ansibleVars: |
+          ctlplane_ip: 192.168.111.101
+          internal_api_ip: 172.17.0.101
+          storage_ip: 172.18.0.101
+          tenant_ip: 172.19.0.101
+          fqdn_internal_api: edpm-compute-1.example.com
+        ansibleSSHPrivateKeySecret: dataplane-ansible-ssh-private-key-secret
+  roles:
+    edpm-compute:
```
[the remark this answers is not in the document]
Comment (rabi) [reply config/samples/dataplane_v1beta1_openstackdataplane_baremetal.yaml:35]: I think they're defaulted and this template has been tested. 

[1] https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/api/v1beta1/common.go#L77

[the remark this answers is not in the document]
Comment (fultonj) [reply config/samples/dataplane_v1beta1_openstackdataplane_baremetal.yaml:35]: I see, thanks.

**Comment by rabi** [comment]
recheck

**Comment by rabi** [comment]
recheck

**Pull request merged**